fe/fe-core/src/main/java/org/apache/doris/job/extensions/insert/InsertJob.java

```diff
@@ -65,58 +100,171 @@ public class InsertJob extends AbstractJob<InsertTask, Map> {
+                     JobStatus jobStatus,
+                     LabelName labelName,
+                     String comment,
+                     UserIdentity createUser,
+                     JobExecutionConfiguration jobConfig,
+                     Long createTimeMs,
+                     String executeSql) {
+        super(jobId, jobName, jobStatus, labelName.getDbName(), comment, createUser,
+                jobConfig, createTimeMs, executeSql, null);
+        this.dbId = ConnectContext.get().getCurrentDbId();
+        this.labelName = labelName.getLabelName();
+    }
 
-    // max save task num, do we need to config it?
-    private static final int MAX_SAVE_TASK_NUM = 100;
+    public InsertJob(ConnectContext ctx,
+                      StmtExecutor executor,
+                      String labelName,
+                      List<InsertIntoTableCommand> plans,
+                      Set<String> sinkTableNames,
+                      Map<String, String> properties,
+                      String comment,
+                      JobExecutionConfiguration jobConfig) {
+        super(Env.getCurrentEnv().getNextId(), labelName, JobStatus.RUNNING, null,
+                comment, ctx.getCurrentUserIdentity(), jobConfig);
+        this.ctx = ctx;
+        this.plans = plans;
+        this.stmtExecutor = executor;
+        this.dbId = ctx.getCurrentDbId();
+        this.labelName = labelName;
+        this.tableNames = sinkTableNames;
+        this.properties = properties;
+        // TODO: not support other type yet
+        this.loadType = InsertJob.LoadType.BULK;
+    }
 
     @Override
-    public List<InsertTask> createTasks(TaskType taskType, Map taskContext) {
-        //nothing need to do in insert job
-        InsertTask task = new InsertTask(null, getCurrentDbName(), getExecuteSql(), getCreateUser());
-        task.setJobId(getJobId());
-        task.setTaskType(taskType);
-        task.setTaskId(Env.getCurrentEnv().getNextId());
-        ArrayList<InsertTask> tasks = new ArrayList<>();
-        tasks.add(task);
-        super.initTasks(tasks);
-        addNewTask(task.getTaskId());
-        return tasks;
+    public List<InsertTask> createTasks(TaskType taskType, Map<Object, Object> taskContext) {
+        if (plans.isEmpty()) {
+            InsertTask task = new InsertTask(labelName, getCurrentDbName(), getExecuteSql(), getCreateUser());
+            task.setTaskType(taskType);
+            task.setJobId(getJobId());
+            task.setCreateTimeMs(System.currentTimeMillis());
+            task.setStatus(TaskStatus.PENDING);
+            ArrayList<InsertTask> tasks = new ArrayList<>();
+            tasks.add(task);
+            super.initTasks(tasks);
+            addNewTask(task.getTaskId());
+            return tasks;
+        } else {
+            return createBatchTasks(taskType);
+        }
     }
 
-    public void addNewTask(long id) {
+    private List<InsertTask> createBatchTasks(TaskType taskType) {
```

Comment:
We should unify the `createBatchTasks()` and `createTasks()`.
For example, why there is `idToTasks.put(task.getTaskId(), task);` in `createBatchTasks()` but not in `createTasks()`?

Reply:
refactor in the future, because there are many conflicts for this method